### ISSUE TITLE
Cannot downgrade from 1.8 to 1.7 - upgrade_downgrade test failure

### ISSUE THREAD

**Issue by su225** [istio/istio#29439]
**Bug description**
```
+ istioctl_path=to_dir.wWzyIX/istio-1.7-alpha.35ffee31deb75f25d1a40f08ce177c41f07773d3/bin
+ to_dir.wWzyIX/istio-1.7-alpha.35ffee31deb75f25d1a40f08ce177c41f07773d3/bin/istioctl upgrade --skip-confirmation --charts to_dir.wWzyIX/istio-1.7-alpha.35ffee31deb75f25d1a40f08ce177c41f07773d3/manifests
Control Plane - ingressgateway pod - istio-ingressgateway-7b4fb74fb9-8lwpc - version: 1.8.0
Control Plane - pilot pod - istiod-f54df64c5-rqdlm - version: 1.8.0

2020-12-06T04:02:59.154218Z	info	Error: upgrade version check failed: 1.8.0 -> 1.7.0. Error: upgrade is currently not supported from version: 1.8.0

Error: upgrade version check failed: 1.8.0 -> 1.7.0. Error: upgrade is currently not supported from version: 1.8.0
+ die 'upgrade failed'
+ echo 'upgrade failed'
upgrade failed
```
https://storage.googleapis.com/istio-prow/logs/istio-downgrade-using-istioctl-1.8_latest-1.7_latest/1335433798547935232/build-log.txt

Upgrade-downgrade test failure. Cannot downgrade from 1.8.0 to 1.7.0

Actually, this job has been failing since at least 27 Oct. Although it shows success, on checking logs, we can see that there is an error while downgrading - https://storage.googleapis.com/istio-prow/logs/istio-downgrade-using-istioctl-1.8_latest-1.7_latest/1/build-log.txt

**Affected product area (please put an X in all that apply)**

[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Comment by su225** [comment]
https://prow.istio.io/job-history/gs/istio-prow/logs/istio-downgrade-using-istioctl-1.8_latest-1.7_latest
It came to light because after merging https://github.com/istio/tools/pull/1303 because we fail if `istioctl upgrade` fails.

**Comment by morvencao** [comment]
This should be fixed in https://github.com/istio/istio/pull/29191

**Comment by su225** [comment]
@morvencao - Downgrading from 1.9 to 1.8 is also failing
```
+ istioctl_path=to_dir.rizUe1/istio-1.8-alpha.b64e3f6100f096124c300e15749b3a799d3b984a/bin
+ to_dir.rizUe1/istio-1.8-alpha.b64e3f6100f096124c300e15749b3a799d3b984a/bin/istioctl upgrade --skip-confirmation --charts to_dir.rizUe1/istio-1.8-alpha.b64e3f6100f096124c300e15749b3a799d3b984a/manifests
2020-12-07T08:02:51.284223Z	info	proto: tag has too few fields: "-"
Control Plane - ingressgateway pod - istio-ingressgateway-56fdb6cdf8-jjljx - version: 1.9.0
Control Plane - pilot pod - istiod-54d9744f65-kz5vq - version: 1.9.0

Upgrade version check passed: 1.9.0 -> 1.8.0.

2020-12-07T08:02:51.903638Z	info	Error: failed to generate Istio configs from file [] for the current version: 1.9.0, error: failed to fetch URL https://github.com/istio/istio/releases/download/1.9.0/istio-1.9.0-osx.tar.gz : 404 Not Found

Error: failed to generate Istio configs from file [] for the current version: 1.9.0, error: failed to fetch URL https://github.com/istio/istio/releases/download/1.9.0/istio-1.9.0-osx.tar.gz : 404 Not Found
```
https://prow.istio.io/job-history/gs/istio-prow/logs/istio-downgrade-using-istioctl-master-1.8_latest
Today's run - https://prow.istio.io/job-history/gs/istio-prow/logs/istio-downgrade-using-istioctl-master-1.8_latest

**Comment by shamsher31** [comment]
Like I said https://github.com/istio/istio/issues/29297#issuecomment-736592242 downgrading from the un-official release will fail. Maybe we should skip downgrading from an un-official version?

**Comment by su225** [comment]
@morvencao - I tried it today and still same problem
```
$ to_dir.oqzcbv/istio-1.7-alpha.74a8d16a80064245a7e31cc884cb6e6e02b08777/bin/istioctl upgrade -y --charts to_dir.oqzcbv/istio-1.7-alpha.74a8d16a80064245a7e31cc884cb6e6e02b08777/manifests/
Control Plane - ingressgateway pod - istio-ingressgateway-6f487d4656-r54fb - version: 1.8.0
Control Plane - pilot pod - istiod-6574485ddb-skn9c - version: 1.8.0

2020-12-15T05:01:49.187329Z	info	Error: upgrade version check failed: 1.8.0 -> 1.7.0. Error: upgrade is currently not supported from version: 1.8.0

Error: upgrade version check failed: 1.8.0 -> 1.7.0. Error: upgrade is currently not supported from version: 1.8.0
```

@shamsher31 - sorry. I forgot that. But it is happening even when both are official releases (here, 1.8.0 and 1.7.0).  This is today's run and it failed again - https://storage.googleapis.com/istio-prow/logs/istio-downgrade-using-istioctl-1.8_latest-1.7_latest/1338695343969144832/build-log.txt

**Comment by shamsher31** [comment]
@su225 The fixed is just [cherry-picked for `1.8`](https://github.com/istio/istio/pull/29288). We also need to cherry-pick the same for `1.7` as downgrading to a lower version requires `1.7` binary. I will do a manual cherry-pick for `1.7`. Thanks.

**Comment by su225** [comment]
https://prow.istio.io/job-history/gs/istio-prow/logs/istio-downgrade-using-istioctl-1.8_latest-1.7_latest
Job history indicates that this is fixed. (It was failing for some other reason and that is fixed as well). So closing this.
